Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/DccV1Parser.kt

```diff
@@ -124,11 +126,32 @@ class DccV1Parser @Inject constructor(
         }
     }
 
+    private fun String.checkSchema(mode: Mode) = when (mode) {
+        Mode.CERT_VAC_STRICT,
+        Mode.CERT_SINGLE_STRICT,
+        Mode.CERT_REC_STRICT,
+        Mode.CERT_TEST_STRICT -> dccJsonSchemaValidator.isValid(this).let {
+            if (it.isValid) return@let this
+            throw InvalidHealthCertificateException(
+                errorCode = ErrorCode.JSON_SCHEMA_INVALID,
+                IllegalArgumentException("Schema Validation did not pass:\n${it.invalidityReason}")
+            )
+        }
+        Mode.CERT_REC_LENIENT,
+        Mode.CERT_TEST_LENIENT,
+        Mode.CERT_VAC_LENIENT -> {
+            // We don't check schema in lenient mode, it may affect already stored certificates.
+            this
+        }
+    }
+
```

Comment:
This is the new check.

We don't need to return `this` and could just use `apply`, but by using the return value, we get a compiler check forcing us to make `when` exhaustive.